test/test_dataloader.py

```diff
@@ -220,6 +253,28 @@ def _test_timeout():
         finally:
             p.terminate()
 
+    def test_worker_seed(self):
+        num_workers = 6
+        dataset = SynchronizedSeedDataset(num_workers, num_workers)
```

Comment:
Here is the added test, the `SynchronizedSeedDataset` is a dataset with a barrier like mechanism baked in. I avoided `Barrier` as it is not available in python 2.